src/Commands/Site/Upstream/SetCommand.php

```diff
@@ -31,10 +31,12 @@ public function set($site_name, $upstream_id)
         if (!$this->confirm('Are you sure you want change the upstream for {site} to {upstream}?', $msg_params)) {
             return;
         }
-        $this->log()->info(
-            'To undo this change run `terminus site:upstream:set {site} {upstream}`',
-            ['site' => $site->id, 'upstream' => $site->getUpstream()->id,]
-        );
+        if (!is_null($previous_upstream_id = $site->getUpstream()->id)) {
```

Comment:
Assignment inside an is_null gives me the heebie jeebies. Also is would a `0`, `''` or `false` be valid here? It might be better to just check for the truthiness of `$previous_upstream_id`

Reply:
FWIW: Assignments inside conditionals are on a lot of people's no-no list but I'm ok with them in PHP as long as they're very simple. I do now tend to take the assignment out of the `if` in deference to other people's sensibilities

Reply:
I know, and we do it everywhere in Terminus already.